src/i18_bluesky/plans/align_lookup_1.py

```diff
@@ -0,0 +1,116 @@
+from dataclasses import dataclass
+from typing import Optional
+
+import bluesky.plan_stubs as bps
+import numpy as np
+import sympy as sp
+from dodal.common import MsgGenerator, inject
+from dodal.devices.undulator import Undulator
+
+monochromator = inject("monochromator")
+
+I0 = inject("i0")
+IT = inject("it")
+t1x = inject("t1x")
+t1y = inject("t1y")
+t1theta = inject("t1theta")
+# note the theta variant is for tomography
+
+pinhole = inject("pinhole")
+
+# get the gaussian shape IT transmission detector we get the shape
+
+
+def derivative_max(data):
+    x, y = sp.symbols("x y")
+    y = sp.Function("y")(x)
+    derivative = sp.diff(y, x)
+    argmax = sp.solve(sp.Eq(derivative, 0), x)
+    return argmax
+
+
+@dataclass
+class BeamlineAlignmentParams:
+    target_energy: Optional[float] = 12.0
+    tolerance: Optional[float] = 0.1
+
```

Comment:
```suggestion
@dataclass
class BeamlineAlignmentParams:
    target_energy: float = 12.0
    tolerance: float = 0.1

```

These fields aren't actually optional on the dataclass: they are optional when you call the dataclass constructor, because they have defaults already set. i.e. calling BeamlineAlignmentParams(target_energy=None) shouldn't be valid, which is what you have implied.